tap_slack/transform.py

```diff
@@ -30,16 +30,13 @@ def transform_json(stream, data, date_fields, channel_id=None):
                 record.pop("parent_conversation", None)
                 record.pop("channel_id", None)
 
-            # Set the date field value for all streams
-            # Set record["thread_ts"] = record.get(date_field, None) in "messages" stream
-            # Set record["thread_ts"] to first data record date_field value in "threads" stream,
-            # so record["thread_ts"] equals to first message ts in thread
+            # Set date field value for all streams
+            # Set "thread_ts" to record value in "messages" stream
+            # Do not change "thread_ts" value for other streams
             for date_field in date_fields:
                 timestamp = record.get(date_field, None)
                 if timestamp and isinstance(timestamp, str):
                     if stream == "messages":
                         record["thread_ts"] = timestamp
-                    elif stream == "threads" and date_field == "ts":
-                        record["thread_ts"] = data[0].get("thread_ts", None)
                     record[date_field] = decimal_timestamp_to_utc_timestamp(timestamp)
```

Comment:
This must work fine.
So we don't touch `thread_ts` at all if it's not `messages` stream and just leave is it.

[log.txt](https://github.com/airbytehq/tap-slack/files/6503239/log.txt) is example of `messages` and  `threads` output records for chat (see screenshots). I've created a slack app to test it properly.
![image](https://user-images.githubusercontent.com/14863647/118698874-9812e380-b819-11eb-905a-d71e699f05d8.png)
![image](https://user-images.githubusercontent.com/14863647/118699043-bf69b080-b819-11eb-8247-619fe84f813e.png)
![image](https://user-images.githubusercontent.com/14863647/118699111-d4deda80-b819-11eb-8140-ec8917716b04.png)